src/mr_reduction/filter_events.py

```diff
@@ -365,77 +326,263 @@ def filter_cross_sections(
+        Minimum number of events required in the workspace. Default is 0.
+    use_slow_flipper_log : bool, optional
+        Whether to use the slow flipper log for filtering. Default is False.
+
+    Returns
+    -------
+    TODO: finish this when i figure out the return type
     """
-    if (file_path is None) and (input_workspace is None):
-        raise ValueError("Either file_path or input_workspace must be provided")
 
     if (file_path is not None) and file_path.endswith(".nxs"):
-        return load_legacy_cross_Sections(file_path, output_workspace)
+        # Handle legacy (pre-EPICS) data files differently
+        xs_list = load_legacy_cross_sections(file_path, output_workspace or mtd.unique_hidden_name())
+        return xs_list
+
+    events_workspace = get_workspace(input_workspace, file_path)
+
+    if events_workspace.getNumberEvents() < min_event_count:
+        raise ValueError(
+            f"Insufficient number of reflected beam events: {input_workspace.getNumberEvents()} "
+            f"(Minimum of {min_event_count} events required)"
+        )
+
+    run_number = int(events_workspace.getRunNumber())
+    if output_workspace is None:
+        output_workspace = str(run_number)
+
+    if use_slow_flipper_log:
+        xs_list = slow_filter_cross_sections(events_workspace)
     else:
-        # if user provides both file_path and input_workspace, use the input_workspace
-        if input_workspace is None:
-            # load data file into a temporary workspace
-            events_workspace = LoadEventNexus(Filename=file_path, OutputWorkspace=mtd.unique_hidden_name())
-        else:
-            events_workspace = workspace_handle(input_workspace)
-        filter_cross_sections(
+        xs_list = filter_cross_sections(
             events_workspace,
             output_workspace,
-            pv_polarizer_state=pv_polarizer_state,
-            pv_analyzer_state=pv_analyzer_state,
-            pv_polarizer_veto=pv_polarizer_veto,
-            pv_analyzer_veto=pv_analyzer_veto,
+            pv_polarizer_state=polarization_logs.POL_STATE,
+            pv_analyzer_state=polarization_logs.ANA_STATE,
+            pv_polarizer_veto=polarization_logs.POL_VETO,
+            pv_analyzer_veto=polarization_logs.ANA_VETO,
             check_devices=check_devices,
         )
-        if input_workspace is None:
+
+        # Only cleanup if the input workspace was created here and is not referenced by the output workspaces
+        if (input_workspace is None) and not (any(ws.name() == str(events_workspace) for ws in xs_list)):
             AnalysisDataService.remove(str(events_workspace))
-        return workspace_handle(output_workspace)
+
+        # If we have no cross section info, treat the data as unpolarized and use Off_Off as the label.
+        for ws in xs_list:
+            if "cross_section_id" not in ws.getRun():
+                ws.getRun()["cross_section_id"] = "Off_Off"
+
+    return xs_list
+
+
+def get_err_list(
```

Comment:
if you change the name of `get_xs_list`, let's make this name follow the same pattern, e.g. `split_events` and `split_error_events`